### PR TITLE
Integrate popgym

### PR DESCRIPTION
**Pull request by nickpetrocelli**
Some TODOs left in that I need feedback on, in particular with regards to testing; wasn't entirely sure what to test in terms of environments so currently my tests just check that the integration works (i.e. environments run without erroring).

While this PR specifically integrates POPGym, this infrastructure should allow us to add any arbitrary Gym environment with minimal effort so long as they are wrapped in a GammaTerminalWrapper. I've updated some function signatures to reflect this.

### CODE REVIEW

**Review by taodav — CHANGES_REQUESTED** [review]
Overall great job on the PR!!!!

I've left a few comments and some suggestions for changes. The biggest being removing my implementation for `gamma_terminal` (including the argument into `step`) and just using the `GammaTerminalWrapper`.

Another "big" change would be the integration tests with POPGym. I think we might have to do a bit of digging and choose two environments (one discrete, one continuous observation spaces) and test expected optimal policy. We (as in both you and I) should take a look at the POPGym paper and maybe choose an environment where GRU can easily learn the optimal policy. I'll take a look at the paper again tomorrow for this.

Otherwise, this is looking great!

File: grl/environment/wrappers/gamma_terminal.py
Comment (nickpetrocelli) [reply grl/environment/wrappers/gamma_terminal.py:18]: working on the above comment, but this shouldn't be a bug (and is in fact how it was originally implemented in `MDP`) because it doesn't touch `terminal` in the case we don't gamma-terminate. In other words, if `step()` already returned `terminal=True`, then the rest of the function will have no effect.

File: tests/test_rnn.py
Comment (nickpetrocelli) [reply tests/test_rnn.py:264]: For the input encoding thing, the weirdness I was commenting on is that the observation space for the _original_ POPGym environment is a `Discrete(4)`, i.e. only enough observations for each possible input value, not "one of four values and a remember indicator" as the POPGym paper claims. I don't think it's an issue with our code.

**Comment by taodav** [comment]
Another thing that I think we should address before we merge this in: action spaces. Currently we only support discrete action spaces, but POPGym has environments which may contain both multi-discrete (so think a vector of ints) or continous action spaces. The POPGym paper actually uses PPO as it's base algorithm, which is more general than algorithm (Sarsa) which we've been using.

**Pull request merged**
